src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -125,9 +125,13 @@ using IriOrUriExpression = NARY<1, FV<std::identity, IriOrUriValueGetter>>;
 
 // STRLEN
 [[maybe_unused]] auto strlen = [](std::string_view s) {
-  return Id::makeFromInt(static_cast<int64_t>(s.size()));
+  // Count UTF-8 characters by skipping continuation bytes (those starting with
+  // "10").
+  auto utf8Len = std::count_if(s.begin(), s.end(), [](char c) {
+    return (static_cast<unsigned char>(c) & 0xC0) != 0x80;
+  });
+  return Id::makeFromInt(static_cast<int64_t>(utf8Len));
```

Comment:
```suggestion
  return Id::makeFromInt(utf8Len);
```
(addresses two of the three suggestions from sonarcloud. `count_if` already returns a signed integer (much to my surprse)

Reply:
It seems like you have forgotten the second suggestion,or was there any particular reason against it?